graph/State-Associated-Subsets.md

```diff
@@ -0,0 +1,113 @@
+# Pattern Name
+
+Microsoft Graph API Design Pattern
+
+
+
+Modeling state associated to a collection that may include all instances, an included subset, an excluded subset, no instances, or any combinations of the above.
+
+## Context
+
+A common pattern is to apply a policy or state to a collection of objects. With this, there also comes the question of how to model cases where we want to apply to 'all' or 'none' as well without having to special case these values within the collection set or introducing cross property dependencies. Likewise we'd like to model it in a way where it is easy to understand and interpret usage from just looking at the schema.
+
+A simple example is where you have some policy you need to be able to apply to users in an organization. You may want to support default **None**, enablement for **All** or enablement for **Select** users where you only grant it to a handful of users.
+
+## Problem
+--------
+
+Existing patterns for this either have special-cased 'strings' or have tightly coupled dependencies between two independent properties. Neither are intuitive, require reading documentation and cannot be inferred from the schema or within client libraries.
```

Comment:
The problem statement should be expressing the problem that is being solved by this pattern, not discussing the disadvantages of other patterns that try to solve this problem.